Core/clim-core/presentation-defs.lisp

```diff
@@ -1657,32 +1657,51 @@ protocol retrieving gestures from a provided string."))
       ;; Now prune out all completions that don't start with the string
       (let ((type (pathname-type actual-pathname)))
 	(when (null type)
-	  ;; If the user didn't supply a file type, don't burden him with all
-	  ;; sorts of version numbers right now.
-	  (let ((new-completions nil))
-	    (dolist (pathname completions)
-	      (cond
-		;; meaning this is actually a directory
-		((and (null (pathname-name pathname))
-		      (null (pathname-type pathname)))
-		 (pushnew (make-pathname :host (pathname-host original-pathname)
-					 :device (pathname-device original-pathname)
-					 :directory (butlast (pathname-directory pathname))
-					 :name (first (last (pathname-directory pathname)))
-					 :type nil)
-			  new-completions))
-		(t
-		 (pushnew (make-pathname :host (pathname-host original-pathname)
-					 :device (pathname-device original-pathname)
-					 :directory (pathname-directory original-pathname)
-					 :name (pathname-name pathname)
-					 :type (pathname-type pathname))
-			  new-completions))))
-	    (setq completions (nreverse new-completions))))
+	  (flet ((legitimate-logical-pathname (name)
+		   (let ((word (string name)))
+		     (dotimes (i (length word))
+		       (let ((ch (schar word i)))
+			 (unless (and (typep ch 'standard-char)
+				      (or (alpha-char-p ch) (digit-char-p ch) (char= ch #\-)))
+			   (return-from legitimate-logical-pathname nil))))
```

Comment:
please use loop and more specific predicates, i.e

```
(loop for ch across (string name)
      unless (and (standard-char-p ch)
                  (alphanumericp ch))
        return nil
      finally (return t))
```

Reply:
```lisp
(loop …
      unless (and …)
        return nil
     finally (return t))
```
can be simplified to `(loop ... always (and …))`.

Reply:
I've accepted you suggestions and will resubmit soon.
Daniel, with regard to tabs, the easier emacs fix is to select the function and then use mx detabify

In terms of an example to illustrate the problem:
1) Make a directory, say "foo"
2) create a file in it with a character not allowed in a logical pathname, e.g. foo_bar
3) create a logical pathname that resolves to that directory

The following code will do that:
(ensure-directories-exist "~/foobar/bar/")
(let* ((final-pathname (pathname "~/foobar/**/*.*")))
  (setf (logical-pathname-translations "foo")
	`(("**;*.*"	,final-pathname))))
(with-open-file (f "~/foobar/bar/baz_bar.lisp" :direction :output :if-does-not-exist :create)
  )
(with-open-file (f "~/foobar/bar/baz-bar.lisp" :direction :output :if-does-not-exist :create)
  )

With the patch doing (clim:accept 'clim:pathname) and responding foo:ba followed by tab (or whatever the complete gesture is) will work, but won't without it.